xclim/indices/fwi.py

```diff
@@ -140,18 +140,26 @@ def day_length(lat, mth):  # pragma: no cover
         dl = DAY_LENGTHS[3, :]
     elif 90 >= lat >= 30:
         dl = DAY_LENGTHS[4, :]
+    elif lat > 90 or lat < -90:
+        raise ValueError("Invalid lat specified.")
+    else:
+        raise ValueError
     return dl[mth - 1]
 
 
 @jit
-def day_length_factor(lat, mth):  # pragma: no cover
+def day_length_factor(lat: Union[int, float], mth: int):  # pragma: no cover
```

Comment:
`int` is implied in `float`, no ? So I don't think you need this union.

Reply:
Yes, will get on that.